### PR TITLE
Add --masonry-view-card-margin

### PR DESCRIPTION
**Pull request by KTibow**
## Proposed change

This adds a new variable, `--ha-card-margin`.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
frontend:
  themes:
    test_theme:
      ha-card-margin: 20px
```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: #3929
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by thomasloven** [comment]
I don't like the name. It's not a property of the ha-card, and doesn't apply to all ha-cards.

It says I edited the post, btw. I think I accidentally clicked one of the checkboxes... sorry.

**Comment by KTibow** [comment]
What would be a good name then? `--lovelace-margin`?

**Comment by zsarnett** [comment]
is it for all of lovelace?

It is for the Masonry view only. so --masonry-view-column-margin would be more appropriate no?

**Comment by KTibow** [comment]
Sounds good. That will make it a little more confusing to find, but it's better to have something than nothing.

File: src/panels/lovelace/views/hui-masonry-view.ts [hunk 1]
```diff
@@ -288,7 +288,7 @@ export class MasonryView extends LitElement implements LovelaceViewElement {
 
       .column > * {
         display: block;
-        margin: 4px 4px 8px;
+        margin: var(--ha-card-margin, 4px 4px 8px);
```
Comment (KTibow) [inline src/panels/lovelace/views/hui-masonry-view.ts:291]:
```suggestion
        margin: var(--masonry-view-column-margin, 4px 4px 8px);
```

**Comment by bramkragten** [comment]
What about stacks and the grid card?

**Comment by KTibow** [comment]
As for the grid card, do you mean changing `var(--grid-card-gap, 8px)` to `var(--grid-card-gap, var(--masonry-view-margin, 8px))`? And if stack cards are added, if they can be used in panels, technically it shouldn't have the `masonry` part?

**Comment by bramkragten** [comment]
If you can change the margin of the view cards you should also be able to set the margin of the cards in stacks to make sure they align?

So you should at least introduce a `--stack-card-margin`

File: src/panels/lovelace/views/hui-masonry-view.ts [hunk 2]
```diff
@@ -295,7 +295,7 @@ export class MasonryView extends LitElement implements LovelaceViewElement {
 
       .column > * {
         display: block;
-        margin: 4px 4px 8px;
+        margin: var(--masonry-view-margin, 4px 4px 8px);
```
Comment (bramkragten) [inline src/panels/lovelace/views/hui-masonry-view.ts:298]:
```suggestion
        margin: var(--masonry-view-card-margin, 4px 4px 8px);
```

Comment (bramkragten) [reply src/panels/lovelace/views/hui-masonry-view.ts:298]: It is the margin that is applied to every card.

File: src/panels/lovelace/cards/hui-horizontal-stack-card.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ class HuiHorizontalStackCard extends HuiStackCard {
         }
         #root > * {
           flex: 1 1 0;
-          margin: 0 4px;
+          margin: var(--stack-margin, var(--horizontal-stack-margin, 0 4px));
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-horizontal-stack-card.ts:32]: This should be the other way? horizontal should override general stack?

Also name it `stack-card-margin`

File: src/panels/lovelace/cards/hui-horizontal-stack-card.ts [hunk 4]
```diff
@@ -29,7 +29,7 @@ class HuiHorizontalStackCard extends HuiStackCard {
         }
         #root > * {
           flex: 1 1 0;
-          margin: 0 4px;
+          margin: var(--horizontal-stack-margin, var(--stack-card-margin, 4px 0));
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-horizontal-stack-card.ts:32]:
```suggestion
          margin: var(--horizontal-stack-card-margin, var(--stack-card-margin, 4px 0));
```

File: src/panels/lovelace/cards/hui-vertical-stack-card.ts [hunk 5]
```diff
@@ -29,7 +29,7 @@ class HuiVerticalStackCard extends HuiStackCard {
           height: 100%;
         }
         #root > * {
-          margin: 4px 0 4px 0;
+          margin: var(--vertical-stack-margin, var(--stack-card-margin, 4px 0));
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-vertical-stack-card.ts:32]:
```suggestion
          margin: var(--vertical-stack-card-margin, var(--stack-card-margin, 4px 0));
```

File: src/panels/lovelace/views/hui-masonry-view.ts (hunk 2 above)
Comment (bramkragten) [inline src/panels/lovelace/views/hui-masonry-view.ts:298]:
```suggestion
        margin: var(--masonry-view-card-margin, 4px 4px 8px);
```

File: src/panels/lovelace/cards/hui-horizontal-stack-card.ts [hunk 6]
```diff
@@ -29,7 +29,7 @@ class HuiHorizontalStackCard extends HuiStackCard {
         }
         #root > * {
           flex: 1 1 0;
-          margin: 0 4px;
+          margin: var(--horizontal-stack-card-margin, var(--stack-card-margin, 4px 0));
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-horizontal-stack-card.ts:32]: You changed the default margin here...

Comment (KTibow) [reply src/panels/lovelace/cards/hui-horizontal-stack-card.ts:32]: Ooops, let me fix that...

Comment (KTibow) [inline src/panels/lovelace/cards/hui-horizontal-stack-card.ts:32]:
```suggestion
          margin: var(--horizontal-stack-card-margin, var(--stack-card-margin, 0 4px));
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**